corehq/apps/geospatial/views.py

```diff
@@ -89,6 +89,29 @@ def mapbox_routing_status(request, domain, poll_id):
     return routing_status(poll_id)
 
 
+def routing_status_view(request, domain, poll_id):
+    # Todo; handle HTTPErrors
+    return json_response({
+        'result': routing_status(poll_id)
+    })
+
+
+class CaseDisbursementAlgorithm(BaseDomainView):
+    urlname = "case_disbursement"
+
+    def post(self, request, domain, *args, **kwargs):
+        solver_class = GeoConfig.objects.get(domain=domain).disbursement_solver
+        request_json = json.loads(request.body.decode('utf-8'))
+        poll_id, result = solver_class(request_json).solve()
+        if poll_id is None:
```

Comment:
what will happen when poll_id is None?

Reply:
Javascript will then not poll for results but rather expect the results to be there.